### PR TITLE
feat: Add list validations helper

### PR DESCRIPTION
**Pull request by tiagogandarezff**
WIP: add method to validate lists

Add some more validations methods and unit tests

Add list helper

Update minor version

# PR Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: tests/unit/test_list.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
+import pytest
+
+import checkarg.list as List
+from checkarg.exceptions import ArgumentError
+
+single_int_element = [1]
+
+seven_ints = [1, 2, 3, 4, 5, 6, 7]
```
Comment (tiagogandarezff) [inline tests/unit/test_list.py:8]: use range instead

File: tests/unit/test_list.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+seven_strings = ["one", "two", "three", "four", "five", "six", "seven"]
+seven_mix = [1, True, 3, "four", 5, False, 7.5]
+
+no_repeated_ints = [1, 2, 3]
```
Comment (tiagogandarezff) [inline tests/unit/test_list.py:12]: use range instead

File: tests/unit/test_list.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+no_repeated_string = ["one", "two", "three"]
+no_repeated_mixed = [1, False, 3.5, "four"]
+
+repeated_ints = [1, 2, 3, 2]
```
Comment (tiagogandarezff) [inline tests/unit/test_list.py:16]: use range

File: tests/unit/test_list.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    pytest.param(no_repeated_string, id="No repeated string"),
+    pytest.param(repeated_string, id="Repeated string"),
+]
+
+not_repeated_lists = [
+    pytest.param(no_repeated_ints, id="No repeated ints"),
+    pytest.param(no_repeated_string, id="No repeated strings"),
+    pytest.param(no_repeated_mixed, id="No repeated mix"),
+]
+
```
Comment (tiagogandarezff) [inline tests/unit/test_list.py:38]: 2 lines

File: tests/unit/test_list.py [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+    List.has_length_between(list, 3, 5)
+
+
+@pytest.mark.parametrize("list", lists_with_length_lower_than_3)
+def test_has_length_between__with_length_lower_than_3__raise_exception(list):
+    with pytest.raises(ArgumentError):
+        List.has_length_between(list, 3, 5)
+
+
+@pytest.mark.parametrize("list", lists_with_length_greater_than_5)
```
Comment (tiagogandarezff) [inline tests/unit/test_list.py:145]: check if I can user 2 @pytest.mark.parametrize lines

File: tests/unit/test_list.py [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+
+@pytest.mark.parametrize("list", lists_with_length_lower_than_3)
+def test_is_length_equals__with_length_lower_than_3__raise_exception(list):
+    with pytest.raises(ArgumentError):
+        List.is_length_equals(list, 7)
+
+
+@pytest.mark.parametrize(
+    "list, element",
+    [([1, 2, 3], 2), (["one", "two", "three"], "one"), ([False, "one", 2.5, 5], 2.5)],
```
Comment (tiagogandarezff) [inline tests/unit/test_list.py:165]: use variables

File: tests/unit/test_list.py (hunk 5 above)
Comment (tiagogandarezff) [reply tests/unit/test_list.py:143]: I couldn't see a good approach for this

**Review by dinispeixoto — APPROVED** [review]

**Review by ff-jonasmaia — APPROVED** [review]

**Pull request merged**